### PR TITLE
Improved exception inheritance

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

A lot of the exception code was repetitive.  All of the exceptions that were using `generate_file_location_string` and `summarize_int_list` were modified to inherit from a new base exception class named `InfileError`.

A feature was added to the base class to be able to manually specify where location information should be incorporated into a supplied message.  This is achieved with an "order" argument that corresponds to what information should be placed where formatting "" characters are in the supplied message.

I also noted that a number of the exception tests were put in the wrong test class, so I rearranged them to be in the appropriate classes.

I also created a new exception class named CompoundDoesNotExist in order to provide file context to the exception.

## Affected Issues/Pull Requests

- Resolves no issue
- Merges into #897
- Next PR: #899

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

**Review by lparsons — APPROVED** [review]
Overall, a nice consolidation of code. Some suggestions inline.

NB: Instead of an `order` parameter, consider using an `f-string` with named parameters instead of a format string with generic `%s` placeholders.

File: DataRepo/utils/exceptions.py [hunk 1]
```diff
@@ -16,33 +16,78 @@
...
+    ):
+        self.rownum = rownum
+        self.sheet = sheet
+        self.file = file
+        self.column = column
+        loc = generate_file_location_string(
+            rownum=rownum, sheet=sheet, file=file, column=column
+        )
+        if "%s" not in message:
+            message += "  Location: %s."
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/utils/exceptions.py:31]: - [ ] TODO: Add comment to message addendum

**Comment by hepcat72** [comment]
> NB: Instead of an `order` parameter, consider using an `f-string` with named parameters instead of a format string with generic `%s` placeholders.

I saw that `f` strings were intended to replace `%s` placeholders, but I didn't dig into it.  I hadn't known previously that placeholders were "outdated".  I'll see if I can work out how this sort of thing is supposed to be done now, when I want the values to be supplied (or generated) separately.

File: DataRepo/utils/exceptions.py [hunk 2]
```diff
@@ -16,33 +16,78 @@
...
+        if order is not None:
+            if "loc" not in order and len(order) != 4:
```
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/utils/exceptions.py:33]: See #902

File: DataRepo/utils/exceptions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hepcat72) [reply DataRepo/utils/exceptions.py:31]: See #902

**Pull request merged**